### PR TITLE
[5.1] Get routes from RouteCollection by its method

### PR DESCRIPTION
**Pull request by pespantelis**

### CODE REVIEW

File: src/Illuminate/Routing/RouteCollection.php [hunk 1]
```diff
@@ -261,6 +261,17 @@ public function hasNamedRoute($name)
...
 
     /**
+     * Get a route instance by its method.
+     *
+     * @param  string  $method
+     * @return \Illuminate\Routing\Route|null
+     */
+    public function getByMethod($method)
+    {
+        return isset($this->routes[$method]) ? $this->routes[$method] : null;
```
Comment (JosephSilber) [inline src/Illuminate/Routing/RouteCollection.php]: Or `return Arr::get($this->routes, $method)`

Comment (pespantelis) [inline src/Illuminate/Routing/RouteCollection.php]: @JosephSilber I followed the same style as the lines [282](https://github.com/laravel/framework/pull/10721/files#diff-6200babcb528ffc5681be487bf9446f2R282) ans [293](https://github.com/laravel/framework/pull/10721/files#diff-6200babcb528ffc5681be487bf9446f2R293).

**Comment by taylorotwell** [comment]
This isn't correct. This would not return a single route but rather a collection of routes. I recommend just getting all the routes and filtering them yourself.

**Comment by pespantelis** [comment]
@taylorotwell you are right.. It's about copy/paste demon!

**Comment by christianpv** [comment]
I was looking for this, wanted an array || collection of routes by method name. Wish it was in the framework. Thanks anyway.

**Pull request closed without merging**